pyccel/parser/semantic.py

```diff
@@ -903,14 +904,27 @@ def _handle_kernel(self, expr, func, args, **settings):
                         severity='fatal')
             # TODO : type check the NUMBER OF BLOCKS 'numBlocks' and threads per block 'tpblock'
             if not isinstance(expr.numBlocks, LiteralInteger):
-                errors.report("Invalid Block number parameter for Kernel call",
+                try:
+                    numBlocks = self.scope.variables[expr.numBlocks]
+                    if not isinstance(numBlocks.dtype, NativeInteger):
+                        errors.report("Invalid Block number parameter for Kernel call",
+                        symbol = expr,
+                        severity='error')
+                except KeyError:
```

Comment:
Not IndexError?

Reply:
No, because I'm looking for items in a dictionary of variables in this scope.
I think it would be better if I used method get instead of square braces.

Reply:
Ok great. I was just confused by the error message in that case. I don't see any reason to use `get` instead of square braces. Square braces for dictionaries are completely standard in python. The only reason I thought it was a list/array rather than a dictionary was because you say "invalid block number"